frontend/packages/kubevirt-plugin/integration-tests/tests/models/virtualMachine.ts

```diff
@@ -1,164 +1,49 @@
 /* eslint-disable no-await-in-loop, no-console */
-import { browser, ExpectedConditions as until } from 'protractor';
-import { isLoaded, resourceTitle } from '@console/internal-integration-tests/views/crud.view';
-import {
-  selectDropdownOption,
-  waitForStringNotInElement,
-  resolveTimeout,
-} from '@console/shared/src/test-utils/utils';
+import { browser } from 'protractor';
+import { waitForStringNotInElement } from '@console/shared/src/test-utils/utils';
 import * as vmView from '../../views/virtualMachine.view';
 import { VMConfig, VMImportConfig } from '../utils/types';
 import {
-  PAGE_LOAD_TIMEOUT_SECS,
   VM_BOOTUP_TIMEOUT_SECS,
   VM_MIGRATION_TIMEOUT_SECS,
   VM_ACTION,
   TAB,
   VM_IMPORT_TIMEOUT_SECS,
-  UNEXPECTED_ACTION_ERROR,
-  VM_ACTIONS_TIMEOUT_SECS,
-  VM_STOP_TIMEOUT_SECS,
   VM_STATUS,
-  VMI_ACTION,
 } from '../utils/consts';
 import { detailViewAction, listViewAction } from '../../views/vm.actions.view';
-import { nameInput as cloneDialogNameInput } from '../../views/dialogs/cloneVirtualMachineDialog.view';
 import { ProvisionConfigName } from '../utils/constants/wizard';
 import { Wizard } from './wizard';
-import { appHost, testName } from '@console/internal-integration-tests/protractor.conf';
-import { KubevirtDetailView } from './kubevirtDetailView';
 import { ImportWizard } from './importWizard';
 import { VirtualMachineModel } from '../../../src/models/index';
+import { BaseVirtualMachine } from './baseVirtualMachine';
 
-const noConfirmDialogActions: (VM_ACTION | VMI_ACTION)[] = [VM_ACTION.Start, VM_ACTION.Clone];
-
-export class VirtualMachine extends KubevirtDetailView {
-  constructor(config, kind?: 'virtualmachines' | 'virtualmachineinstances') {
-    super({ ...config, kind: kind || 'virtualmachines' });
-  }
-
-  async getStatus(): Promise<string> {
-    return vmView.vmDetailStatus(this.namespace, this.name).getText();
-  }
-
-  async getNode(): Promise<string> {
-    return vmView.vmDetailNode(this.namespace, this.name).getText();
-  }
+// TODO: Remove VM_ACTION.Delete action from the list when BZ 1827640 if resolved
+const noConfirmDialogActions: VM_ACTION[] = [VM_ACTION.Start, VM_ACTION.Clone, VM_ACTION.Delete];
 
-  async getBootDevices(): Promise<string[]> {
-    return vmView.vmDetailBootOrder(this.namespace, this.name).getText();
+export class VirtualMachine extends BaseVirtualMachine {
+  constructor(config) {
+    super({ ...config, kind: VirtualMachineModel });
   }
 
-  async action(action: VM_ACTION | VMI_ACTION, waitForAction?: boolean, timeout?: number) {
+  async action(action: VM_ACTION, waitForAction?: boolean, timeout?: number) {
     await this.navigateToTab(TAB.Details);
 
-    const confirmDialog = !noConfirmDialogActions.includes(action);
-
-    await detailViewAction(action, confirmDialog);
+    await detailViewAction(action, !noConfirmDialogActions.includes(action));
     if (waitForAction !== false) {
```

Comment:
I know this isn't included in this PR, but if you make any changes, could you please change this to `if (waitForAction)`? Unless I'm missing something, `waitForAction` shouldn't require the `!== false` comparison.